### PR TITLE
HDPI-5751-Screen changes Notice date Known-FE

### PR DESCRIPTION
**Pull request by sdmiddletonSolirius**
### Jira link

See [HDPI-5751](https://tools.hmcts.net/jira/browse/HDPI-5751)

**Related Backend PR:** https://github.com/hmcts/pcs-api/pull/2116

### Change description

- Added notice document link and service method to the "Notice date" confirmation page -- when a possession notice has been uploaded, the page now renders a "View a copy of the notice" link (opening the document in a new tab) and a bullet point describing the method of service (e.g. "by first class post", "by other means: Carrier Pigeon").
- Extended `extractCaseDocuments` utility and `viewDocuments` route to resolve notice documents from `notice_Documents` and `detailsTab_NoticeDetails.noticeDocuments` in addition to `allDocuments`, since the possession notice is excluded from the generic allDocuments list.
- Added CCD model fields and TypeScript interface entries for `notice_ServiceMethod`, `notice_PersonName`, `notice_EmailAddress`, `notice_OtherExplanation`, `notice_OtherElectronicExplanation`, `notice_Documents`, and `detailsTab_NoticeDetails` to support the notice details on the citizen read path.
- Added English and Welsh locale entries for the notice link text and all six service method descriptions, with Welsh placeholder translations.
- added unit tests for notice document resolution from both source fields, service method text rendering for all six methods, edge cases (missing/unknown method, empty documents, able-to-upload with no doc), and `documentUtils` extraction from `notice_Documents` and `detailsTab_NoticeDetails`.

**Notice by First Class Post**:

notice-by-first-class-post

**Notice by personally handing it**:

notice-by-personally-handing-it

**Notice by Email**:

notice-by-email

**Notice by other electronic means**:

notice-by-other-electronic-means

**Notice by Other means**:

notice-by-other-means

**Notice to a permitted place**:

notice-to-permitted-place

### Security Vulnerability Assessment ###

**CVE Suppression:** Are there any CVEs present in the codebase (either newly introduced or pre-existing) that are being intentionally suppressed or ignored by this commit?
  * [ ] Yes
  * [ ] No

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/routes/viewDocuments.ts [hunk 1]
```diff
@@ -88,7 +88,17 @@ export default function viewDocumentsRoutes(app: Application): void {
           id?: string;
           value?: { document_filename?: string; document_binary_url?: string };
         }[];
-        const document = allDocuments.find(item => item.id === documentId)?.value;
+        const noticeDocuments = (ccdCase.data?.notice_Documents ?? []) as {
```
Comment (andrewdami11) [inline src/main/routes/viewDocuments.ts]: Could this handler use `findCaseDocumentById(ccdCase.data ?? {}, documentId)` now that `extractCaseDocuments` already covers `allDocuments`, `notice_Documents`, and `detailsTab_NoticeDetails.noticeDocuments`? That would avoid keeping parallel lookup logic in sync?

File: src/main/steps/respond-to-claim/confirmation-of-notice-date-when-provided/index.ts [hunk 2]
```diff
@@ -109,12 +111,62 @@ export const step: StepDefinition = createRespondToClaimFormStep({
...
+    const formatNoticeDate = (raw?: string): string =>
+      raw ? DateTime.fromISO(raw).setZone('Europe/London').setLocale('en-gb').toFormat('d LLLL y') : '';
+
+    const serviceMethod = validatedCase?.notice_ServiceMethod;
+    let noticeMethodText: string | undefined;
+
+    switch (serviceMethod) {
+      case 'PERSONALLY_HANDED':
+        noticeMethodText = t('methodOfService.PERSONALLY_HANDED', {
+          name: validatedCase?.notice_PersonName ?? '',
```
Comment (andrewdami11) [inline src/main/steps/respond-to-claim/confirmation-of-notice-date-when-provided/index.ts]: If `notice_PersonName`, `notice_EmailAddress`, or the explanation fields are missing, is it worth omitting `noticeMethodText` entirely rather than interpolating empty strings into `methodOfService.*`?

File: src/main/steps/respond-to-claim/confirmation-of-notice-date-when-provided/index.ts [hunk 3]
```diff
@@ -109,12 +111,62 @@ export const step: StepDefinition = createRespondToClaimFormStep({
...
+    const caseData = (validatedCase?.data as Record<string, unknown>) ?? {};
+    const documents = extractCaseDocuments(caseData);
+    const noticeDoc =
+      documents.find(d => d.sourceField === 'detailsTab_NoticeDetails.noticeDocuments') ??
+      documents.find(d => d.sourceField === 'notice_Documents');
+    const noticeDocumentId = noticeDoc?.id;
+    const noticeDocumentFilename = noticeDoc?.filename;
+
+    const formatNoticeDate = (raw?: string): string =>
+      raw ? DateTime.fromISO(raw).setZone('Europe/London').setLocale('en-gb').toFormat('d LLLL y') : '';
```
Comment (andrewdami11) [inline src/main/steps/respond-to-claim/confirmation-of-notice-date-when-provided/index.ts]: Should `formatNoticeDate` use the request locale (for example via `req.language` / existing `formatDate` helpers) instead of hard-coding `en-gb`, so `DELIVERED_PERMITTED_PLACE` dates localise with the Welsh strings?

File: src/test/unit/routes/viewDocuments.test.ts [hunk 4]
```diff
@@ -347,5 +347,55 @@ describe('viewDocuments route', () => {
       stream.emit('error', new Error('stream failed'));
       expect(next).toHaveBeenCalledWith(expect.objectContaining({ message: 'Failed to stream document', status: 502 }));
     });
+
+    it('streams document binary from detailsTab_NoticeDetails.noticeDocuments when not in allDocuments', async () => {
```
Comment (andrewdami11) [inline src/test/unit/routes/viewDocuments.test.ts:398]: Could we also add a route test that streams from `notice_Documents` when `allDocuments` and `detailsTab_NoticeDetails` are empty, matching the citizen event-trigger path?

File: src/main/services/ccdCaseData.model.ts [hunk 5]
```diff
@@ -97,6 +98,30 @@ export class CcdCaseModel {
...
+
+  get notice_OtherExplanation(): string | undefined {
+    return this.data.notice_OtherExplanation;
+  }
+
+  get notice_OtherElectronicExplanation(): string | undefined {
+    return this.data.notice_OtherElectronicExplanation;
+  }
+
+  get noticeDocument(): CcdCollectionItem<Document> | undefined {
```
Comment (andrewdami11) [inline src/main/services/ccdCaseData.model.ts]: Is `noticeDocument` needed on the model? If so, should it mirror the step preference for `detailsTab_NoticeDetails.noticeDocuments` over `notice_Documents`, or can we remove the unused getter?

File: src/main/routes/viewDocuments.ts (hunk 1 above)
Comment (andrewdami11) [reply src/main/routes/viewDocuments.ts]: Addressed here: https://github.com/hmcts/pcs-frontend/commit/304d91e6a

File: src/main/steps/respond-to-claim/confirmation-of-notice-date-when-provided/index.ts (hunk 2 above)
Comment (andrewdami11) [reply src/main/steps/respond-to-claim/confirmation-of-notice-date-when-provided/index.ts]: Addressed here: https://github.com/hmcts/pcs-frontend/commit/6631369e3

File: src/main/steps/respond-to-claim/confirmation-of-notice-date-when-provided/index.ts (hunk 3 above)
Comment (andrewdami11) [reply src/main/steps/respond-to-claim/confirmation-of-notice-date-when-provided/index.ts]: Addressed here: https://github.com/hmcts/pcs-frontend/commit/6631369e3

File: src/test/unit/routes/viewDocuments.test.ts (hunk 4 above)
Comment (andrewdami11) [reply src/test/unit/routes/viewDocuments.test.ts:398]: Addressed here: https://github.com/hmcts/pcs-frontend/commit/304d91e6a

File: src/main/services/ccdCaseData.model.ts (hunk 5 above)
Comment (andrewdami11) [reply src/main/services/ccdCaseData.model.ts]: Addressed here: https://github.com/hmcts/pcs-frontend/commit/6631369e3

**Review by fudge88 — APPROVED** [review]

**Review by sadmanrahh — APPROVED** [review]

**Review by andrewdami11 — APPROVED** [review]
